zone/client_packet.cpp

```diff
@@ -16183,3 +16343,45 @@ void Client::RecordKilledNPCEvent(NPC *n)
 		}
 	}
 }
+
+void Client::Handle_OP_RaidDelegateAbility(const EQApplicationPacket* app) 
+{
+	if (app->size != sizeof(DelegateAbility_Struct)) {
+		LogDebug("Size mismatch in OP_RaidDelegateAbility expected [{}] got [{}]", sizeof(DelegateAbility_Struct), app->size);
+		DumpPacket(app);
+		return;
+	}
+
+	DelegateAbility_Struct* das = (DelegateAbility_Struct*)app->pBuffer;
+
+	switch (das->DelegateAbility)
+	{
+	case RaidDelegateMainAssist: 
+	{
+		auto r = entity_list.GetRaidByName(this->GetName());
+		r->DelegateAbilityAssist(this, das->Name);
+		break;
+	}
+	case RaidDelegateMainMarker: 
+	{
+		auto r = entity_list.GetRaidByName(this->GetName());
+		r->DelegateAbilityMark(this, das->Name);
+		break;
+	}
+	default:
+		LogDebug("RaidDelegateAbility default case");
+		break;
+	}
+}
+
+void Client::Handle_OP_RaidClearNPCMarks(const EQApplicationPacket* app)
+{
+	if (app->size != 0)	{
+		LogDebug("Size mismatch in OP_RaidClearNPCMark expected [{}] got [{}]", 0, app->size);
+		DumpPacket(app);
+		return;
+	}
+
+	auto r = entity_list.GetRaidByName(GetName());
```

Comment:
You have access to Client here, so just do `auto r = GetRaid();`

Reply:
Suggest the following. It validates the client raid object is valid before calling it

```cpp
	auto r = GetRaid();
	if (r) {
		r->RaidClearNPCMarks(this);
	}
```

Reply:
Updated